tikv-client/src/main/java/com/pingcap/tikv/util/ChannelFactory.java

```diff
@@ -52,9 +52,14 @@ public ManagedChannel getChannel(String addressStr) {
 
   @Override
   public void close() {
-    for (ManagedChannel ch : connPool.values()) {
-      ch.shutdown();
-    }
+    connPool.forEach(
+        (k, v) -> {
+          v.shutdownNow();
+          try {
+            v.awaitTermination(1, TimeUnit.SECONDS);
```

Comment:
why wait for 1 second?

Reply:
No particular reason. This is added to make a graceful shutdown.